component/http/v2/check.go

```diff
@@ -0,0 +1,63 @@
+package v2
+
+import (
+	"net/http"
+)
+
+// AliveStatus type representing the liveness of the service via HTTP component.
+type AliveStatus int
+
+// ReadyStatus type.
+type ReadyStatus int
+
+const (
+	// Alive represents a state defining a Alive state.
+	Alive AliveStatus = 1
+	// Unresponsive represents a state defining a Unresponsive state.
+	Unresponsive AliveStatus = 2
+
+	alivePath = "/alive"
+
+	// Ready represents a state defining a Ready state.
+	Ready ReadyStatus = 1
+	// NotReady represents a state defining a NotReady state.
+	NotReady ReadyStatus = 2
+
+	readyPath = "/ready"
+)
+
+// ReadyCheckFunc defines a function type for implementing a readiness check.
+type ReadyCheckFunc func() ReadyStatus
+
+// AliveCheckFunc defines a function type for implementing a liveness check.
+type AliveCheckFunc func() AliveStatus
```

Comment:
Couldn't we use `LivenessCheckFunc`? I know patron is supposed to be k8s-agnostic but still, I feel like it's more common than alive.